### PR TITLE
Smart fields

### LINKED ISSUE #59: Create smart fields in ButtonField.jsx
__Description:__
When inputing a Facebook or mail link, user must type `www.facebook.com/__name` or `mailto:__email`. Instead, this should be auto populated for the user depending on the link type selected by them.

__Requires:__
* Editor component to be stable
* Props to be relatively stable

__Breakdown:__
- [ ] Change prefix in text field based on type prop
- [ ] Create understandable UX for user (interaction shouldn't be surprising)

__Time Estimation:__
1.5 hrs.

### PR DESCRIPTION
**Pull request by hwkr**
Also polishes the add remove buttons

Resolves #59

### CODE REVIEW

File: client/components/parts/ButtonField.jsx [hunk 1]
```diff
@@ -33,13 +34,56 @@ export default class ButtonField extends Component{
...
     const kind = e.target.dataset.kind
     const val = this.props.value
-    val[kind] = e.target.value
+    if (kind === 'type'){
+      // Swap prefixes
+      const oldtemplate = this.props.typeTemplates[val.type]
+      const newtemplate = this.props.typeTemplates[e.target.value]
+      val.href = val.href.replace(new RegExp(`^(${oldtemplate.preprefix}|${oldtemplate.prefix})+`), '')
+      if (newtemplate.prefix) val.href = newtemplate.prefix + val.href
+      if (newtemplate.preprefix) val.href = newtemplate.preprefix + val.href
```
Comment (bndnio) [inline client/components/parts/ButtonField.jsx:80]: Add some comments here explaining what's going on -- hard to understand the purpose of pre/prepre-fixes and the regex

File: client/components/parts/ButtonField.jsx [hunk 2]
```diff
@@ -33,13 +34,56 @@ export default class ButtonField extends Component{
...
+        preprefix: 'http://',
+        prefix: 'twitter.com/'
+      },
+      instagram: {
+        label: 'Instagram Account',
+        placeholder: 'something',
+        preprefix: 'http://',
+        prefix: 'instagram.com/'
+      }
+    }
```
Comment (bndnio) [inline client/components/parts/ButtonField.jsx:66]: Looks good with defaults here 👍

File: client/components/parts/ButtonField.jsx [hunk 3]
```diff
@@ -33,13 +34,56 @@ export default class ButtonField extends Component{
...
+    ],
+    typeTemplates: {
+      link: {
+        label: 'Link URL',
+        placeholder: 'http://www.somewhere.com'
+      },
+      email: {
+        label: 'Email Address',
+        preprefix: 'mailto:',
+        placeholder: 'person@somewhere.com'
```
Comment (bndnio) [inline client/components/parts/ButtonField.jsx:46]: why a preprefix but no prefix? 
Maybe better language could be chosen for this part of the templates?

**Review by bndnio — DISMISSED** [review]

File: client/components/parts/ButtonField.jsx (hunk 3 above)
Comment (hwkr) [reply client/components/parts/ButtonField.jsx:46]: Suggestions?

**Pull request merged**